packages/core-components/src/components/ProgressBars/Gauge.tsx

```diff
@@ -104,19 +119,40 @@ export const getProgressColor: GaugePropsGetColor = ({
  */
 
 export function Gauge(props: GaugeProps) {
+  const hoverRef = useRef<HTMLDivElement>(null);
```

Comment:
I think callback-based `ref` is a bit more robust here. For example

```tsx
const [hoverRef, setHoverRef] = useState<HTMLDivElement | null>(null)
// ...
<div ref={setHoverRef} ...
```

That way you can add `hoverRef` as a dependency to `useEffect` as well.

Bit more info over [here](https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node)